answerking_app/tests/test_unit/test_unit_productserializer.py

```diff
@@ -0,0 +1,211 @@
+    test_cat_data: dict = utb.get_fixture(
+        "categories", "burgers_cat_data.json"
+    )
+    test_prod_data: dict = utb.get_fixture(
+        "products", "plain_burger_data.json"
+    )
+    test_product_serializer: ProductSerializer = ProductSerializer(
+        test_prod_data
+    )
+
+    def setUp(self):
+        cat: Category = Category.objects.create(**self.test_cat_data)
+        prod: Product = Product.objects.create(**self.test_prod_data)
+        cat.products.add(prod)
+
+    def tearDown(self):
+        Category.objects.all().delete()
+        Product.objects.all().delete()
+
+    def test_product_serializer_contains_correct_fields(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: list[str] = [
+            "id",
+            "name",
+            "description",
+            "price",
+            "categories",
+            "retired",
+        ]
+        actual: list[str] = list(test_serializer_data.keys())
+        self.assertCountEqual(actual, expected)
+
+    def test_product_serializer_id_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: int = test_prod.id
+        actual: int = test_serializer_data["id"]
+        self.assertEqual(actual, expected)
+
+    def test_product_serializer_name_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: str = test_prod.name
+        actual: str = test_serializer_data["name"]
+        self.assertEqual(actual, expected)
+
+    def test_product_serializer_description_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: str = test_prod.description
+        actual: str = test_serializer_data["description"]
+        self.assertEqual(actual, expected)
+
+    def test_product_serializer_price_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        test_serializer_data: ReturnDict = ProductSerializer(test_prod).data
+        expected: Decimal = test_prod.price
+        actual: Decimal = test_serializer_data["price"]
+        self.assertEqual(actual, expected)
+
+    def test_product_serializer_categories_field_content(self):
+        test_prod: Product = Product.objects.get(
+            name=self.test_prod_data["name"]
+        )
+        category: Category = test_prod.category_set.first()
```

Comment:
Instead of this I would probably do categories = test_prod.category_set.all() and then do category = categories[0]. Then at the end check that len(categories) == 1.

That way you are checking that 1 category exists and you know that you are getting the correct one. The current way you do it there could be 2 categories and you wouldn't know.

Reply:
Done